File: java/07_auth_failures.java
// A07 - Auth Failures. VULN: missing MFA, long-lived JWT without verify.
import java.util.Base64;

public class AuthFailures {
  public static String mintToken(String username) {
    String payload = "{\"sub\":\"" + username + "\",\"admin\":true}"; // VULN: trust client claims
    return Base64.getEncoder().encodeToString(payload.getBytes());
  }

  public static boolean verify(String token) {
    return token != null && token.length() > 10; // VULN: no signature check
  }
}

File: javascript/07_auth_failures.js
// A07 - Auth Failures. VULN: JWT "none" alg + cookie without HttpOnly.
// VULN: A07 - accepts alg:none tokens
function parseJwt(token) {
  const [header, payload] = token.split(".");
  const h = JSON.parse(Buffer.from(header, "base64url"));
  if (h.alg === "none") return JSON.parse(Buffer.from(payload, "base64url"));
  return null;
}

function setSessionCookie(res, token) {
  res.setHeader("Set-Cookie", `session=${token}`); // VULN: no HttpOnly/Secure
}

module.exports = { parseJwt, setSessionCookie };
